resources/perftest/perftest.cpp

```diff
@@ -0,0 +1,150 @@
+#include <iostream>
+#include <chrono>
+#include <functional>
+#include <random>
+#include <string>
+#include <algorithm>
+
+#include <openassetio/specification/Specification.hpp>
+#include <openassetio/trait/BlobTrait.hpp>
+using namespace openassetio;
+
+constexpr size_t nIterations = 100000;
+const size_t dbSize = 1000000;
+
+std::chrono::nanoseconds MeasureRuntime(std::function<void(void)> func)
+{
+  const auto start = std::chrono::high_resolution_clock::now();
+  func();
+  const auto stop = std::chrono::high_resolution_clock::now();
+  return stop - start;
+}
+
+auto MeasureRuntimeMs(std::function<void(void)> func)
+{
+  return std::chrono::duration_cast<std::chrono::milliseconds>(MeasureRuntime(func)).count();
+}
+
+///////////////////////////////////////////////////////////
+std::string getEntityReference(size_t i) 
+{
+  return std::string("ams://myasset_") + std::to_string(i);
```

Comment:
[consider][minor] It looks like we're right on the borderline of [small string optimisation](https://shaharmike.com/cpp/std-string/) here.  Perhaps for reproducibility we should increase the minimum length of the string to eliminate that as a factor.  To be fair, it probably doesn't make a difference.

Reply:
Addressed in https://github.com/TheFoundryVisionmongers/OpenAssetIO/pull/270/commits/cd4528dc2f32d3c6f86fd0294f3e169d17fcc170

This changes the measurements slightly:
```
resolveEntityReference: 51 ms // was 45 ms before
resolve: 138 ms // was 125 ms before
```